pkg/varcontext/builder.go

```diff
@@ -79,8 +80,8 @@ func (builder *ContextBuilder) MergeDefaults(brokerVariables []DefaultVariable)
 			continue
 		}
 
-		if strVal, ok := v.Default.(string); ok {
-			builder.MergeEvalResult(v.Name, strVal, v.Type)
+		if strVal, ok := v.Default.(string); ok && strVal == "" && v.Expression != "" {
```

Comment:
This part of logic must be reimplemented, does not cover all cases (unit tests are failing) but for the cloud storage service - it is working fine